### ISSUE TITLE
How to search for optimal hyperparameters with TimeGAN (continued)?

### ISSUE THREAD

**Issue by rdvhoorn** [jsyoon0823/TimeGAN#68]
Thanks for the response on issue  #67. I'd like to ask a couple follow-up questions:

1. What fidelity metric did you use for the hyperparameter optimization? In the paper, you mentioned a discriminative score for testing the fidelity. If you greedily optimize for a discriminative score, wouldn't you just get to the first set of parameters which mode-collapses? If it mode-collapses, you will get very good results for such a discriminative score but the model would be of very bad quality. Did you somehow detect mode-collapse algorithmically to prevent the greedy search from choosing those settings? Or did you have to inspect all the models visually?
2. In the greedy hyper-parameter search, what parameters did you tune? Did you have different learning rates for the different modules (generator, discriminator etc.) separately? Did you have a different number of hidden nodes for the different modules? Did you try different optimizers?
3. What kind of greedy hyper-parameter search did you use? Did you just start at a random parameter set, and then afterwards, change one parameter and check the results for improvement? Why did you not use something like bayesian optimization? 
4. For how many epochs did you train each model before assessing its quality? Did you run a full 50000 epochs for each model in your greedy search? 
5. Do you use the supervisor to generate samples or do you skip the supervisor during inference? 

Thanks in advance for all your time!

**Comment by jsyoon0823** [comment]
1. We use predictive score as the criteria
- Even with discriminative score, mode collapsed model will not be selected because the discriminator can easily identify the mode collapsed model.
2. We can include any hyper-parameters for hyper-parameter tuning.
- But to make it more efficient, we only include some sensitive hyper-parameters including hidden_dim, num_layers, epochs, and iterations.
3. We select some sensitive hyper-parameters and their ranges. Then, we try all the combinations and use the best combination. 
4. Number of epochs is also included in the hyper-parameter tuning (i.e., number of epochs is one hyper-parameter that we need to tune).
5. Supervisor is included at inference time. Please see how X_hat is generated.
Thanks!
